tarok/test/cards_tests.cpp

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2020 Semantic Weights. All rights reserved. */
+
+#include "gtest/gtest.h"
+#include "src/cards.h"
+
+namespace tarok {
+
+TEST(CardsTests, TestDealCards) {
+  auto [talon, private_cards] = DealCards(42);
+
+  EXPECT_EQ(talon.size(), 6);  // first element is talon and has 6 cards
+  EXPECT_EQ(private_cards[0].size(), 16);  // each player has 16 cards
+  EXPECT_EQ(private_cards[1].size(), 16);  // each player has 16 cards
+  EXPECT_EQ(private_cards[2].size(), 16);  // each player has 16 cards
+
+  std::array<std::vector<int>, 4> splits = {talon, private_cards[0],
+                                            private_cards[1], private_cards[2]};
+
+  for (int i = 0; i < splits.size() - 1; i++) {
```

Comment:
This code is a bit complex. Would flattening the talon and private cards (i.e. merging all into a single array of length 54) and then comparing it to a vector of sequential elements from 0 to 53 be equivalent?